### PR TITLE
Experimental new interface

### PR DESCRIPTION
**Pull request by MrPowers**

### CODE REVIEW

File: optimus/df_transformer_exp.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+def trim_col(colName):
+    def inner(df):
+        return df.withColumn(colName, trim(col(colName)))
+    return inner
+
+
+def remove_special_chars(colName):
+    def inner(df):
+        special_chars = ["!", "&", "."]
+        regexp = "|".join(['\{0}'.format(i) for i in special_chars])
```
Comment (pirate) [inline optimus/df_transformer_exp.py]: `[` and `]` not needed inside the `.join()`

File: optimus/df_transformer_exp.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+def trim_col(colName):
+    def inner(df):
+        return df.withColumn(colName, trim(col(colName)))
+    return inner
+
+
+def remove_special_chars(colName):
+    def inner(df):
+        special_chars = ["!", "&", "."]
```
Comment (pirate) [inline optimus/df_transformer_exp.py]: Tuple is lighter, and generally recommended unless you're adding/removing stuff: 
```pythonspecial_chars = ('!', '&', '.')`

File: tests/test_df_transformer_exp.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                ("BOB", 1),
+                ("J oSe", 2)
+            ],
+            [
+                ("name", StringType(), True),
+                ("age", IntegerType(), False)
+            ]
+        )
+
+        assert(expected_df.collect() == actual_df.collect())
```
Comment (pirate) [inline tests/test_df_transformer_exp.py]: no parens for assert, it's a keyword:

```python

File: tests/test_df_transformer_exp.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
+from optimus.spark import get_spark
+from optimus.df_transformer_exp import *
+import optimus as op
+from quinn.extensions import *
+from pyspark.sql.types import *
+from pyspark.sql.functions import lit
+
+
+class TestDfTransformerExp(object):
```
Comment (pirate) [inline tests/test_df_transformer_exp.py]: Don't inherit from `(object)` in python >= 3.

**Comment by FavioVazquez** [comment]
Hi @pirate and @MrPowers! @argenisleon and me have been thinking hard about this PR and we really feel is where Optimus needs to go. 

It's gonna be hardwork because we need to refactor the whole library. Are you willing to help in this tasks? 

Here are several .py files in the project. We may need to refactor the classes too.

**Comment by pirate** [comment]
Ah sorry, don't have the time to jump into another project, I'm not really a pyspark/Optimus user.  I was just adding some minor comments since @MrPowers asked me to code review his design earlier.  Feel free to ignore them though, I shouldn't be backseat driving in a repo I'm not contributing to anyway 😁 .

**Comment by FavioVazquez** [comment]
If you find anytime it could be great that you can help @pirate

**Comment by FavioVazquez** [comment]
Maybe just like reviewing code will be great @MrPowers

**Comment by MrPowers** [comment]
@pirate - Thanks for the code review!!  I pushed up a commit to [make all of your changes](https://github.com/ironmussa/Optimus/pull/177/commits/b07ef21be54591d565ab6a9c429da28e807f660f).  I love code reviews and I'm trying to get better at Python, so feel free to give me comments any time.

@FavioVazquez - I'll ping you with a more detailed reply in the issue 😉

**Comment by MrPowers** [comment]
@FavioVazquez - Take a look at the `reorder_columns` and `remove_chars` methods.  Do those look good to you?  Thanks!

**Comment by FavioVazquez** [comment]
Hi @MrPowers! Those look great. We need more assertions and tests so the user have an idea of what's going on. But great. They are interesting.

**Comment by FavioVazquez** [comment]
Great @MrPowers so this works. I did it in a different way in the other functions. But do you think this is more performant? I like it is more functional. I'll start rewriting the functions to work with this version. I would like to add an experimental tag. I'll look for that. Thanks for the great work :)

**Comment by FavioVazquez** [comment]
Hi @MrPowers can you move this to the original dftransformer file? So I can continue with the migration for the existing functions. Thanks :)

**Pull request closed without merging**